### ISSUE TITLE
Vanilla magic mod can bypass minecolonies protections.

### ISSUE THREAD

**Issue by ProsperCraft** [ldtteam/minecolonies#632]
### Minecolonies version

minecolonies-universal-1.11-0.6.2837

### Steps to reproduce the problem
1. Install Vanilla magic
2. Work through all of the achievements
3. Shift right click with a blaze rod with a book in your offhand while having the proper achievements

**Comment by Sejoslaw** [comment]
If You could write the list of blocks that should be blacklisted I could add some kind of blacklist to this spell.

**Comment by marvin-bitterlich** [comment]
It seems that Vanilla Magic does not respect/trigger events with what it does. 

Many mods do this for efficiency (and well, it's easier that way) but we can't protect against mods that do not trigger events. 

@Sejoslaw A blacklist won't work. You would have to issue events for all changed blocks (and other world interactions) and respect that other mods can cancel these events. Without that no protection system can protect against changes.

**Comment by Sejoslaw** [comment]
I did add the Event that is fired when Player will try to save / load block. Canceling this event should prevent block from being taken.
[Event class](https://github.com/Sejoslaw/VanillaMagic/blob/1.11.2/VanillaMagic/src/main/java/seia/vanillamagic/api/event/EventMoveBlock.java#L63)

**Comment by marchermans** [comment]
@Sejoslaw you should probably fire break and place events when you swap the blocks. Blindly just changing blocks can cause a lot of problems!

**Comment by marchermans** [comment]
Maybe even trigger the respective methods on the blocks that you manipulate.

**Comment by marvin-bitterlich** [comment]
@Sejoslaw @ProsperCraft any news on this?

**Comment by Sejoslaw** [comment]
@Kostronor Is Minecolonies about to be updated to 1.11.2 ?

**Comment by Raycoms** [comment]
there is a 1.11 version already.

2017-04-18 7:40 GMT-03:00 Krzysztof Dobrzyński <notifications@github.com>:

**Comment by Sejoslaw** [comment]
@Kostronor  I did try to run the newest version of Minecolonies and it crashed - [Crash Report](https://pastebin.com/4dBKDiKs)

**Comment by marvin-bitterlich** [comment]
@Sejoslaw that looks bad 😅 
Can you open a separate issue for that? Looks like a build problem

**Comment by Sejoslaw** [comment]
Sure :)

**Comment by marvin-bitterlich** [comment]
Closing, waiting on dev of vanilla magic
